### PR TITLE
add docs for webhook signature

### PR DESCRIPTION
**Pull request by bhalsey**
### Issue:
https://github.com/FusionAuth/fusionauth-issues/issues/1859

### Problem:
Customers need to know how webhook signatures work

### Solution:
Document it

### Linked PR:
https://github.com/FusionAuth/fusionauth-app/pull/322

### Reviewer Notes:
Cross check with `signature.md` in linked PR.

### CODE REVIEW

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:84]: Sometimes for larger examples we put the JavaScript in an example repository and then include it from the raw GitHub URL like some of the WebAuthn stuff:
```
https://raw.githubusercontent.com/FusionAuth/fusionauth-example-javascript-webauthn/main/buffer-to-base64url.js
```

Not sure whether we want to do that with this one or not. I see that there is other JS for writing a webhook that is pretty lengthy.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:151]:
```suggestion
* Signatures validated using a public key (RSA or EC) where signature verification dynamically fetches public key from  `.well-known/jwks.json` endpoint
```

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:149]:
```suggestion
link:/docs/v1/tech/tutorials/key-rotation[Rotating keys] regularly is an important part of a defense-in-depth strategy.  The type of key used for signing webhook events and the method used for fetching that key determines the process for rotating keys.
```

**Review by spwitt — CHANGES_REQUESTED** [review]
Mostly minor verbiage and copy fixes. You may want to get a second opinion before moving sample code to GitHub.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (bhalsey) [reply site/docs/v1/tech/events-webhooks/signing.adoc:1]: old habits die hard

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (bhalsey) [reply site/docs/v1/tech/events-webhooks/signing.adoc:50]: it's genericish config file formatting.  Renders better than `text`.

File: site/docs/v1/tech/apis/_webhook-request-body.adoc
Comment (bhalsey) [reply site/docs/v1/tech/apis/_webhook-request-body.adoc:51]: `webhook.eventsEnabled` is another example of specifying an optional Object.  I think this is technically more correct, but you're right that we don't always list the base object.

File: site/docs/v1/tech/apis/_webhook-response-body-base.adoc
Comment (bhalsey) [reply site/docs/v1/tech/apis/_webhook-response-body-base.adoc:63]: this is the response

File: site/docs/v1/tech/apis/_webhook-response-body-base.adoc
Comment (bhalsey) [reply site/docs/v1/tech/apis/_webhook-response-body-base.adoc:67]: this is the response

File: site/docs/v1/tech/apis/_webhook-response-body-base.adoc
Comment (bhalsey) [reply site/docs/v1/tech/apis/_webhook-response-body-base.adoc:70]: this is the response

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (bhalsey) [reply site/docs/v1/tech/events-webhooks/signing.adoc:84]: good question.

File: site/docs/v1/tech/apis/_webhook-request-body.adoc
Comment (robotdan) [reply site/docs/v1/tech/apis/_webhook-request-body.adoc:51]: We never (as far as I know) document the object - just the properties. 

So in this case, we would document two field:

```
webhook.signatureConfiguration.enabled
webhook.signatureConfiguration.signingKeyId
```

You should be able to compare to any other API with nested fields. Application and Tenant have lots of them.

File: site/docs/v1/tech/apis/_webhook-request-body.adoc
Comment (robotdan) [reply site/docs/v1/tech/apis/_webhook-request-body.adoc:55]: > Required if [field]#webhook.signatureConfiguration# is provided.

This is never required.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (mooreds) [reply site/docs/v1/tech/events-webhooks/signing.adoc:84]: I would love to have this code in a separate repo, maybe even with tests?

File: site/docs/v1/tech/apis/_webhook-request-body.adoc
Comment (bhalsey) [reply site/docs/v1/tech/apis/_webhook-request-body.adoc]: Looks like we only document objects for open ended fields like `webhook.eventsEnabled` and `webhook.data`.

Added since tags and removed the `webhook.signatureConfiguration` object field.

File: site/docs/v1/tech/apis/_webhook-request-body.adoc
Comment (bhalsey) [reply site/docs/v1/tech/apis/_webhook-request-body.adoc]: fixed.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (bhalsey) [reply site/docs/v1/tech/events-webhooks/signing.adoc]: Done!  (without the tests)

File: site/assets/img/docs/webhook-settings-signature.png
Comment (bhalsey) [inline site/assets/img/docs/webhook-settings-signature.png:1]: Signature is now under the Security panel, so this screenshot will need to be redone after certificate key handling code is merged.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:10]: 🦅 👁️ 
```suggestion
Signing Webhook events allows you to verify each event was sent by FusionAuth.  Webhook signatures provide an extra layer of security in addition to the methods described in link:/docs/v1/tech/events-webhooks/securing[Securing a Webhook]. Signature verification provides protection against bad actors spoofing webhook event payloads to look like they came from FusionAuth.
```

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:34]: There aren't details on generating the RSA key pair. It seems like it just jumps from "let's create an RSA key pair" straight to assigning it.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:36]: Based on some other examples, we use `breadcrumb` when referring to tabs in the UI. I would recommend changing up how this refers to UI elements because the enable is not a checkbox, and the select is a dropdown ("select box" sounds to me like the redirect URI fields that you can enter and add multiple values).

```suggestion
Next, you configure your webhook to sign the event with this key. From [breadcrumb]#Settings -> Webhooks#, click on the Edit button for your webhook (or create a new webhook). Select the [breadcrumb]#Signature# tab panel. Once you enable webhook signatures, a select dropdown allows you to choose the generated key.
```

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (spwitt) [inline site/docs/v1/tech/events-webhooks/signing.adoc:79]: Not sure why this phrasing doesn't bother me here. Just confirming that you know this will display `(available on GitHub)` in the parentheses.

File: site/assets/img/docs/webhook-settings-signature.png
Comment (spwitt) [reply site/assets/img/docs/webhook-settings-signature.png:1]: 😿

**Review by spwitt — CHANGES_REQUESTED** [review]
Mainly backticks and rogue double-space. I'm not insistent on verbiage changes. Just a couple of things that read weirdly to me the first time through.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (bhalsey) [reply site/docs/v1/tech/events-webhooks/signing.adoc]: I don't think additional details are really necessary in this guide, but added a link to the Key Master Guide.

File: site/docs/v1/tech/events-webhooks/signing.adoc
Comment (bhalsey) [reply site/docs/v1/tech/events-webhooks/signing.adoc:79]: Yep.  (and agree it was a bit awkward in the other place)

**Review by spwitt — APPROVED** [review]

**Pull request merged**
